File: mailway_ui/lib/data/entity/ChatWithChatMessagesWithChatMemberNameStubs.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:mailwayui/data/entity/Chat.dart';
import 'package:mailwayui/data/entity/ChatMemberNameStub.dart';
import 'package:mailwayui/data/entity/ChatMessage.dart';
import 'package:mailwayui/data/entity/ChatMessageAndQuoteMessage.dart';

part 'ChatWithChatMessagesWithChatMemberNameStubs.g.dart';


@JsonSerializable()
class ChatWithChatMessagesWithChatMemberNameStubs {
  Chat chat;
  List<ChatMessageAndQuoteMessage> messages;
  List<ChatMemberNameStub> chatMemberNameStubs;


  ChatWithChatMessagesWithChatMemberNameStubs({
    this.chat,
    this.messages,
    this.chatMemberNameStubs,
  });

  factory ChatWithChatMessagesWithChatMemberNameStubs.fromJson(Map<String, dynamic> json) =>
      _$ChatWithChatMessagesWithChatMemberNameStubsFromJson(json);

  Map<String, dynamic> toJson() => _$ChatWithChatMessagesWithChatMemberNameStubsToJson(this);
}

File: mailwaycore/src/main/java/com/dimension/mailwaycore/data/entity/ChatWithChatMessagesWithChatMemberNameStubs.kt
package com.dimension.mailwaycore.data.entity

import androidx.room.Embedded
import androidx.room.Junction
import androidx.room.Relation
import kotlinx.serialization.Serializable


@Serializable
data class ChatWithChatMessagesWithChatMemberNameStubs(
    @Embedded val chat: Chat,
    @Relation(
        entity = ChatMessage::class,
        parentColumn = "chatId",
        entityColumn = "chatId"
    )
    val messages: List<ChatMessageAndQuoteMessage>,
    @Relation(
        parentColumn = "chatId",
        entityColumn = "chatMemberNameStubId",
        associateBy = Junction(ChatAndChatMemberNameStubCrossRef::class)
    )
    val chatMemberNameStubs: List<ChatMemberNameStub>
)